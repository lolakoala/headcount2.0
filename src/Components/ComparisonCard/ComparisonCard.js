import React from 'react';
import PropTypes from 'prop-types';
import rightArrow from '../../../assets/right-arrow.png';
import leftArrow from '../../../assets/left-arrow.png';

const ComparisonCard = ({ firstDistrict,
                          firstAvg,
                          secondDistrict,
                          secondAvg,
                          comparedAvg }) => {
  return (
    <div className='comparison-card'>
      <h3 className='first-district'>
        {firstDistrict}: {firstAvg}
      </h3>
      <div className="logo-and-compared">
      <img className="left-arrow"
        src={leftArrow}
           alt='arrow pointing left' />
      <h2 className='compared-avg'>
        {comparedAvg}
      </h2>
      <img className="right-arrow"
           src={rightArrow}
           alt='arrow pointing right' />
         </div>
      <h3 className='second-district'>
        {secondDistrict}: {secondAvg}
      </h3>
    </div>
  );
};

ComparisonCard.propTypes = {
  firstDistrict: PropTypes.string.isRequired,
  firstAvg: PropTypes.number.isRequired,
  secondDistrict: PropTypes.string.isRequired,
  secondAvg: PropTypes.number.isRequired,
  comparedAvg: PropTypes.number.isRequired
};

export default ComparisonCard;
