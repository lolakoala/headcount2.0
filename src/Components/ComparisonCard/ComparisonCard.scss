@import '../css/variables.scss';

.comparison-card {
  color: #334855;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: 2px solid #555555;
  border-radius: 5%;
  box-shadow: 4px 4px 2px #555555;
  width: 18%;
  margin: 5px 5px 5px 5px;
  background-color: #ACD0DD;
}

.logo-and-compared {
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  justify-content: center;
  margin: 30px 30px 30px 30px;
}

.compared-avg {
  padding: 0px 10px 0px 10px;
}

.compare-container {
    display: flex;
    flex-direction: row;
    justify-content: center;
    background-color: $header-background-color;
    margin-top: 40px;
  }
