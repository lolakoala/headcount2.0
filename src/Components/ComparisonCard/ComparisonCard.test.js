import React from 'react';
import ComparisonCard from './ComparisonCard.js';
import { shallow } from 'enzyme';

describe('ComparisonCard', () => {
  let wrapper = shallow(<ComparisonCard firstDistrict='Denver'
                                        firstAvg={0.5}
                                        secondDistrict='Englewood'
                                        secondAvg={0.4}
                                        comparedAvg={0.65}/>);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have first district with avg', () => {
    let firstInfo = wrapper.find('.first-district');

    expect(firstInfo.length).toEqual(1);
    expect(firstInfo.text()).toEqual('Denver: 0.5');
  });

  it('should have second district with avg', () => {
    let secondInfo = wrapper.find('.second-district');

    expect(secondInfo.length).toEqual(1);
    expect(secondInfo.text()).toEqual('Englewood: 0.4');
  });

  it('should have two arrow images', () => {
    let arrows = wrapper.find('img');

    expect(arrows.length).toEqual(2);
  });

  it('should have compared avg', () => {
    let comparedNum = wrapper.find('.compared-avg');

    expect(comparedNum.length).toEqual(1);
    expect(comparedNum.text()).toEqual('0.65');
  });
});
