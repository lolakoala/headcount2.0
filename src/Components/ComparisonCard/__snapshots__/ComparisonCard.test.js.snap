// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonCard should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="comparison-card"
>
    <h3
        className="first-district"
    >
        Denver
        : 
        0.5
    </h3>
    <img
        alt="arrow pointing left"
        src="http://www.iconsdb.com/icons/preview/purple/arrow-left-xl.png"
    />
    <h2
        className="compared-avg"
    >
        0.65
    </h2>
    <img
        alt="arrow pointing right"
        src="http://www.iconsdb.com/icons/preview/purple/arrow-right-xl.png"
    />
    <h3
        className="second-district"
    >
        Englewood
        : 
        0.4
    </h3>
</div>,
  "nodes": Array [
    <div
      className="comparison-card"
>
      <h3
            className="first-district"
      >
            Denver
            : 
            0.5
      </h3>
      <img
            alt="arrow pointing left"
            src="http://www.iconsdb.com/icons/preview/purple/arrow-left-xl.png"
      />
      <h2
            className="compared-avg"
      >
            0.65
      </h2>
      <img
            alt="arrow pointing right"
            src="http://www.iconsdb.com/icons/preview/purple/arrow-right-xl.png"
      />
      <h3
            className="second-district"
      >
            Englewood
            : 
            0.4
      </h3>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ComparisonCard
        comparedAvg={0.65}
        firstAvg={0.5}
        firstDistrict="Denver"
        secondAvg={0.4}
        secondDistrict="Englewood"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "comparedAvg": 0.65,
          "firstAvg": 0.5,
          "firstDistrict": "Denver",
          "secondAvg": 0.4,
          "secondDistrict": "Englewood",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="comparison-card"
>
          <h3
                    className="first-district"
          >
                    Denver
                    : 
                    0.5
          </h3>
          <img
                    alt="arrow pointing left"
                    src="http://www.iconsdb.com/icons/preview/purple/arrow-left-xl.png"
          />
          <h2
                    className="compared-avg"
          >
                    0.65
          </h2>
          <img
                    alt="arrow pointing right"
                    src="http://www.iconsdb.com/icons/preview/purple/arrow-right-xl.png"
          />
          <h3
                    className="second-district"
          >
                    Englewood
                    : 
                    0.4
          </h3>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="comparison-card"
>
          <h3
                    className="first-district"
          >
                    Denver
                    : 
                    0.5
          </h3>
          <img
                    alt="arrow pointing left"
                    src="http://www.iconsdb.com/icons/preview/purple/arrow-left-xl.png"
          />
          <h2
                    className="compared-avg"
          >
                    0.65
          </h2>
          <img
                    alt="arrow pointing right"
                    src="http://www.iconsdb.com/icons/preview/purple/arrow-right-xl.png"
          />
          <h3
                    className="second-district"
          >
                    Englewood
                    : 
                    0.4
          </h3>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ComparisonCard
    comparedAvg={0.65}
    firstAvg={0.5}
    firstDistrict="Denver"
    secondAvg={0.4}
    secondDistrict="Englewood"
/>,
}
`;
